core/src/impl/Kokkos_Profiling_Interface.hpp

```diff
@@ -54,6 +54,34 @@
 #include <iostream>
 #include <cstdlib>
 
+// NOTE: in this Kokkos::Profiling block, do not define anything that shouldn't
+// exist should Profiling be disabled
+
+namespace Kokkos {
+namespace Profiling {
+enum struct DeviceType {
+  Serial,
+  OpenMP,
+  Cuda,
+  HIP,
+  OpenMPTarget,
+  HPX,
+  ROCm,
+  Threads
+};
+template <typename ExecutionSpace>
+struct DeviceTypeTraits;
+
+constexpr const size_t device_type_bits = 8;
+constexpr const size_t instance_bits    = 24;
+template <typename ExecutionSpace>
+inline uint32_t device_id(ExecutionSpace const& space) noexcept {
```

Comment:
Please you guys discuss what is the appropriate name for that function too

Reply:
Christian and I did, and this is what we decided. Did you have a different idea? I'd love to hear it